File: maxhanna.client/src/services/datacontracts/nexus/nexus-attack-sent.ts
export interface NexusAttackSent {
  id: number;
  originCoordsX: number;
  originCoordsY: number;
  originUserId?: number;
  destinationCoordsX: number;
  destinationCoordsY: number;
  destinationUserId?: number; 
  marineTotal: number;
  goliathTotal: number;
  siegeTankTotal: number;
  scoutTotal: number;
  wraithTotal: number;
  battlecruiserTotal: number;
  glitcherTotal: number;
  timestamp: Date;
  duration: number;
  arrived?: boolean;
}

File: maxhanna.Server/Controllers/DataContracts/Nexus/NexusAttackSent.cs
using maxhanna.Server.Controllers.DataContracts.Users;

namespace maxhanna.Server.Controllers.DataContracts.Nexus
{
    public class NexusAttackSent
    { 
        public int Id { get; set; }
        public int OriginCoordsX { get; set; }
        public int OriginCoordsY { get; set; }
        public User? OriginUser { get; set; }
        public int DestinationCoordsX { get; set; }
        public int DestinationCoordsY { get; set; }
        public User? DestinationUser { get; set; }
        public int? MarineTotal { get; set; }
        public int? GoliathTotal { get; set; }
        public int? SiegeTankTotal { get; set; }
        public int? ScoutTotal { get; set; }
        public int? WraithTotal { get; set; }
        public int? BattlecruiserTotal { get; set; }
        public int? GlitcherTotal { get; set; }
        public int Duration { get; set; }
        public DateTime Timestamp{ get; set; }
        public bool? Arrived { get; set; }
    }
}